sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala

```diff
@@ -40,18 +40,34 @@ case class UnresolvedNamespace(multipartIdentifier: Seq[String]) extends LeafNod
   override def output: Seq[Attribute] = Nil
 }
 
+abstract class UnresolvedTableOrIdentifierClause extends LeafNode
+
 /**
  * Holds the name of a table that has yet to be looked up in a catalog. It will be resolved to
  * [[ResolvedTable]] during analysis.
  */
 case class UnresolvedTable(
     multipartIdentifier: Seq[String],
     commandName: String,
-    relationTypeMismatchHint: Option[String]) extends LeafNode {
+    relationTypeMismatchHint: Option[String]) extends UnresolvedTableOrIdentifierClause {
   override lazy val resolved: Boolean = false
 
   override def output: Seq[Attribute] = Nil
 }
+/**
+ * Holds the identifier of a tabl ein String form that has yet to be looked up in a catalog.
+ * It will be resolved to [[UnResolvedTable]] during analysis.
+ */
+case class UnresolvedTableIdentifierClause(
```

Comment:
should this just be in `unresolved.scala` next to the other new unresolved identifier clause nodes? Same for the others below.

Reply:
I wondered why the related nodes (such as UnresolvedView) are in this file. If we move the new objects to unresolved.scala, we should move the existing ones as well. @cloud-fan @gengliangwang ?